pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -254,13 +258,158 @@ func (n *NetworkPolicyController) deleteNamespace(old interface{}) {
+	if reflect.DeepEqual(oldServiceAccount.Labels, curServiceAccount.Labels) {
+		klog.V(2).InfoS("Labels of ServiceAccount didn't change, so skip ACNP reprocess", "ServiceAccountNS", curServiceAccount.Namespace, "ServiceAccountName", curServiceAccount.Name, "ServiceAccountLabels", curServiceAccount.Labels)
+		return
+	}
+	klog.V(2).InfoS("Processing ServiceAccount UPDATE event", "ServiceAccountNS", curServiceAccount.Namespace, "ServiceAccountName", curServiceAccount.Name, "OldServiceAccountLabels", oldServiceAccount.Labels, "CurServiceAccountLabels", curServiceAccount.Labels)
+	affectedACNPsByOldServiceAccount := n.filterACNPsByServiceAccount(oldServiceAccount)
+	affectedACNPsByCurServiceAccount := n.filterACNPsByServiceAccount(curServiceAccount)
+	getACNPNamesSet := func(acnps []*crdv1alpha1.ClusterNetworkPolicy) sets.String {
+		nameSet := sets.String{}
+		for _, acnp := range acnps {
+			nameSet.Insert(acnp.Name)
+		}
+		return nameSet
+	}
+	oldACNPNameSet := getACNPNamesSet(affectedACNPsByOldServiceAccount)
+	curACNPNameSet := getACNPNamesSet(affectedACNPsByCurServiceAccount)
+	affectedACNPNameSet := utilsets.SymmetricDifferenceString(oldACNPNameSet, curACNPNameSet)
+	for _, acnp := range affectedACNPsByOldServiceAccount {
+		if affectedACNPNameSet.Has(acnp.Name) {
+			n.reprocessCNP(acnp, false)
+		}
+	}
+	for _, acnp := range affectedACNPsByCurServiceAccount {
+		if affectedACNPNameSet.Has(acnp.Name) {
+			n.reprocessCNP(acnp, false)
+		}
+	}
+}
+
+// deleteNamespace receives ServiceAccount DELETE events and triggers all
+// ClusterNetworkPolicies that refer to this ServiceAccount in `serviceAccounts`
+// field to be re-processed.
+func (n *NetworkPolicyController) deleteServiceAccount(old interface{}) {
+	serviceAccount := old.(*v1.ServiceAccount)
+	defer n.heartbeat("deleteServiceAccount")
+	klog.V(2).InfoS("Processing ServiceAccount DELETE event", "ServiceAccountNS", serviceAccount.Namespace, "ServiceAccountName", serviceAccount.Name, "ServiceAccountLabels", serviceAccount.Labels)
+	affectedACNPs := n.filterACNPsByServiceAccount(serviceAccount)
+	for _, acnp := range affectedACNPs {
+		n.reprocessCNP(acnp, false)
+	}
+}
+
 // processClusterNetworkPolicy creates an internal NetworkPolicy instance
 // corresponding to the crdv1alpha1.ClusterNetworkPolicy object. This method
 // does not commit the internal NetworkPolicy in store, instead returns an
 // instance to the caller wherein, it will be either stored as a new Object
 // in case of ADD event or modified and store the updated instance, in case
 // of an UPDATE event.
 func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *crdv1alpha1.ClusterNetworkPolicy) *antreatypes.NetworkPolicy {
+	// Translate ServiceAccounts to PodSelector+NamespaceSelector may cause
+	// Spec.AppliedTo and Rule.AppliedTo are both nil, when ServiceAccounts don't
+	// exist or they haven't been used by any Pods. Check this before process an ACNP.
+	if !n.validateAppliedToWithSA(cnp) {
+		return nil
```

Comment:
I think this event deserve logging, unless there is already some other routine doing that.

Reply:
Good one. Added.